apollo/integrations/db/motherduck_proxy_client.py

```diff
@@ -0,0 +1,41 @@
+import os
+import duckdb
+from typing import (
+    Any,
+    Dict,
+    Optional,
+)
+
+from apollo.integrations.db.base_db_proxy_client import BaseDbProxyClient
+
+_ATTR_CONNECT_ARGS = "connect_args"
+
+
+class MotherDuckProxyClient(BaseDbProxyClient):
+    """
+    Proxy client for Motherduck client. Credentials are expected to be supplied under "connect_args" and will
+    be passed directly to `duckdb.connect`. 'duckdb` accepts a connection string with the connection details,
+    so "connect_args" will be a string.
+    """
+
+    def __init__(self, credentials: Optional[Dict], **kwargs: Any):
+        if not credentials or _ATTR_CONNECT_ARGS not in credentials:
+            raise ValueError(
+                f"Motherduck agent client requires {_ATTR_CONNECT_ARGS} in credentials"
+            )
+
+        # Duckdb can be used in-memory or to connect to Motherduck, a serverless duckdb.
+        # Duckdb isn't smart enough for us to specify we only care about connecting to
+        # the cloud. It looks for the env var 'HOME' to know where to set up local files
+        # in case you use it in-memory. Lambda functions don't have a HOME env var so we
```

Comment:
Have we confirmed that this approach works on GCP and Azure too? I ask because you explicitly mentioned lambda

Reply:
I agree, we should test this for Azure and GCP.
Also, I think it would be safer to define `HOME` only if not defined yet, I'm pretty sure it's going to be defined for Azure considering they use a Debian image, and maybe also for GCP

Reply:
I did test with all 3 agent flavors. All 3 did work but GCP and AWS needed the cloudrun/lambda memory increased in order to work. Azure worked without any changes.

I updated the logic to only create HOME env var if it is not already set